QSB/RespawnOnDeath.cs

```diff
@@ -0,0 +1,110 @@
+            QSB.Helper.HarmonyHelper.AddPrefix<DeathManager>("KillPlayer", typeof(Patches), nameof(Patches.PreFinishDeathSequence));
+        }
+
+        void PlayerWokeUp()
+        {
+            _playerSpawner = FindObjectOfType<PlayerSpawner>();
+            _fluidDetector = Locator.GetPlayerCamera().GetComponentInChildren<FluidDetector>();
+            _playerResources = Locator.GetPlayerTransform().GetComponent<PlayerResources>();
+
+            var shipTransform = Locator.GetShipTransform();
+            if (shipTransform)
+            {
+                _shipComponents = Locator.GetShipTransform().GetComponentsInChildren<ShipComponent>();
+                _shipBody = Locator.GetShipBody();
+                _shipSpawnPoint = GetSpawnPoint(true);
+
+                // Move debug spawn point to initial ship position.
+                _playerSpawnPoint = GetSpawnPoint();
+                _shipSpawnPoint.transform.position = shipTransform.position;
+                _shipSpawnPoint.transform.rotation = shipTransform.rotation;
+            }
+
+        }
+
+        public void ResetShip()
+        {
+            if (!_shipBody)
+            {
+                return;
+            }
+
+            // Reset ship position.
+            _shipBody.SetVelocity(_shipSpawnPoint.GetPointVelocity());
+            _shipBody.WarpToPositionRotation(_shipSpawnPoint.transform.position, _shipSpawnPoint.transform.rotation);
+
+            // Reset ship damage.
+            if (Locator.GetShipTransform())
+            {
+                for (int i = 0; i < _shipComponents.Length; i++)
+                {
+                    _shipComponents[i].SetDamaged(false);
+                }
+            }
+        }
+
+        public void ResetPlayer()
+        {
+            // Reset player position.
+            OWRigidbody playerBody = Locator.GetPlayerBody();
+            playerBody.WarpToPositionRotation(_playerSpawnPoint.transform.position, _playerSpawnPoint.transform.rotation);
+            playerBody.SetVelocity(_playerSpawnPoint.GetPointVelocity());
+            _playerSpawnPoint.AddObjectToTriggerVolumes(Locator.GetPlayerDetector().gameObject);
+            _playerSpawnPoint.AddObjectToTriggerVolumes(_fluidDetector.gameObject);
+            _playerSpawnPoint.OnSpawnPlayer();
+
+            _playerResources.SetValue("_isSuffocating", false);
+
+            // Reset player health and resources.
+            _playerResources.DebugRefillResources();
+        }
+
+        SpawnPoint GetSpawnPoint(bool isShip = false)
+        {
+            var spawnList = _playerSpawner.GetValue<SpawnPoint[]>("_spawnList");
+            for (int i = 0; i < spawnList.Length; i++)
+            {
+                SpawnPoint spawnPoint = spawnList[i];
+                if (spawnPoint.GetSpawnLocation() == SpawnLocation.TimberHearth && spawnPoint.IsShipSpawn() == isShip)
+                {
+                    return spawnPoint;
+                }
+            }
+            return null;
```

Comment:
return _playerSpawner.GetValue<SpawnPoint[]>("_spawnList").FirstOrDefault(spawnPoint => spawnPoint.GetSpawnLocation() == SpawnLocation.TimberHearth && spawnPoint.IsShipSpawn() == isShip);
You don't have to do this, just saying 😁